src/std/io/bio/api.ss

```diff
@@ -39,7 +40,7 @@
 (def (open-buffered-writer writer (buffer-size default-buffer-size))
   (unless (Writer? writer)
     (error "Expected Writer instance" writer))
-  (BufferedWriter (make-output-buffer write (make-u8vector buffer-size) 0)))
+  (BufferedWriter (make-output-buffer writer (make-u8vector buffer-size) 0)))
```

Comment:
Btw this should be the kind of error spotted by the new `gsc` `-warnings` flag.

Reply:
interesting

Reply:
with contracts (also planned for v18), maybe the gerbil compiler will also be able to catch it.